src/openrct2/ride/Vehicle.cpp

```diff
@@ -836,116 +838,146 @@ static void vehicle_invalidate(rct_vehicle * vehicle)
-                            v += 16 * vehicle->var_BF;
-                            i->frequency = (uint16)v;
-                            i->id        = vehicle->sprite_index;
-                            i->volume    = 0;
-                            if (vehicle->x != LOCATION_NULL)
-                            {
-                                rct_tile_element * tile_element = map_get_surface_element_at(vehicle->x >> 5, vehicle->y >> 5);
-                                if (tile_element != NULL && tile_element->base_height * 8 > vehicle->z)
-                                { // vehicle underground
-                                    i->volume = 0x30;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
+    uint16 sound_priority = vehicle_get_sound_priority_factor(vehicle);
+    rct_vehicle_sound_params * soundParam;
+    // Find a sound param of lower priority to use
+    for (soundParam = &gVehicleSoundParamsList[0];
+        soundParam < gVehicleSoundParamsListEnd && sound_priority <= soundParam->priority;
+        soundParam++)
+        ;
+
+    if (soundParam >= &gVehicleSoundParamsList[Util::CountOf(gVehicleSoundParamsList)])
+        return;
+
+    if (gVehicleSoundParamsListEnd < &gVehicleSoundParamsList[Util::CountOf(gVehicleSoundParamsList)])
+    {
+        gVehicleSoundParamsListEnd++;
+    }
+
+    // Shift all sound params down one if using a free space
+    if (soundParam != gVehicleSoundParamsListEnd)
+    {
+        Memory::Move<rct_vehicle_sound_params>(
+            soundParam + 1,
+            soundParam,
+            ((gVehicleSoundParamsListEnd - soundParam) - 1) * sizeof(rct_vehicle_sound_params));
+    }
+
+    soundParam->priority = sound_priority;
+    sint32 pan_x =
+        (vehicle->sprite_left / 2) + (vehicle->sprite_right / 2) - g_music_tracking_viewport->view_x;
+    pan_x >>= g_music_tracking_viewport->zoom;
+    pan_x += g_music_tracking_viewport->x;
+
+    uint16 screenwidth = context_get_width();
+    if (screenwidth < 64)
+    {
+        screenwidth = 64;
+    }
+    soundParam->pan_x = ((((pan_x * 65536) / screenwidth) - 0x8000) >> 4);
+
+    sint32 pan_y =
+        (vehicle->sprite_top / 2) + (vehicle->sprite_bottom / 2) - g_music_tracking_viewport->view_y;
+    pan_y >>= g_music_tracking_viewport->zoom;
+    pan_y += g_music_tracking_viewport->y;
+
+    uint16 screenheight = context_get_height();
+    if (screenheight < 64)
+    {
+        screenheight = 64;
+    }
+    soundParam->pan_y = ((((pan_y * 65536) / screenheight) - 0x8000) >> 4);
+
+    sint32 frequency = std::abs(vehicle->velocity);
+
+    rct_ride_entry * ride_type = get_ride_entry(vehicle->ride_subtype);
+    if (ride_type != NULL)
+    {
+        if (ride_type->vehicles[vehicle->vehicle_type].double_sound_frequency & 1)
+        {
+            frequency *= 2;
+        }
+    }
+
+    frequency >>= 5;
+    frequency *= 5512;
```

Comment:
Do those number mean anything particular?

Reply:
At the moment no. I expect it's an approximated number.